packages/eas-cli/src/commands/branch/list.ts

```diff
@@ -18,6 +18,59 @@ import { findProjectRootAsync, getProjectFullNameAsync } from '../../project/pro
 import { getActorDisplayName } from '../../user/actions';
 
 const BRANCHES_LIMIT = 10_000;
+export async function listBranchesAsync({
+  fullName,
+}: {
+  fullName: string;
+}): Promise<
+  (Pick<UpdateBranch, 'id' | 'name'> & {
+    updates: (Pick<Update, 'id' | 'updatedAt' | 'message'> & {
+      actor?: Maybe<Pick<User, 'username' | 'id'> | Pick<Robot, 'firstName' | 'id'>>;
+    })[];
+  })[]
```

Comment:
Not sure what to make of this type. It seems like something that's been copy pasted from the generated types and it's bound to change whenever the query below is changed.

I believe TypeScript could actually infer this return type without having it written out here. I know we have a convention to explicitly write the return types in functions, but I think in this case it would be better / more maintainable to let TypeScript infer the return type.

Reply:
<img width="647" alt="Screen Shot 2021-03-09 at 17 16 17" src="https://user-images.githubusercontent.com/497214/110492983-33415b80-80fb-11eb-81e3-aa9d0a0d6657.png">

If you remove the return type, it's still inferred correctly.